fvm/handler/contract.go

```diff
@@ -169,58 +266,59 @@ func (h *ContractHandler) Commit() ([]programs.ContractUpdateKey, error) {
 	return updatedKeys, nil
 }
 
-func (h *ContractHandler) Rollback() error {
-	h.draftUpdates = make(map[programs.ContractUpdateKey]programs.ContractUpdate)
-	return nil
-}
-
-func (h *ContractHandler) HasUpdates() bool {
-	return len(h.draftUpdates) > 0
+func (updater *contractUpdater) Reset() {
+	updater.draftUpdates = make(map[programs.ContractUpdateKey]programs.ContractUpdate)
 }
 
-type contractUpdateKeyList []programs.ContractUpdateKey
-
-func (l contractUpdateKeyList) Len() int      { return len(l) }
-func (l contractUpdateKeyList) Swap(i, j int) { l[i], l[j] = l[j], l[i] }
-func (l contractUpdateKeyList) Less(i, j int) bool {
-	switch bytes.Compare(l[i].Address[:], l[j].Address[:]) {
-	case -1:
-		return true
-	case 0:
-		return l[i].Name < l[j].Name
-	default:
-		return false
-	}
+func (updater *contractUpdater) hasUpdates() bool {
+	return len(updater.draftUpdates) > 0
 }
 
-func (h *ContractHandler) UpdateKeys() []programs.ContractUpdateKey {
-	if len(h.draftUpdates) == 0 {
-		return nil
+func (updater *contractUpdater) updates() (
+	programs.ContractUpdateKeyList,
+	programs.ContractUpdateList,
+) {
+	if len(updater.draftUpdates) == 0 {
+		return nil, nil
 	}
-	keys := make(contractUpdateKeyList, 0, len(h.draftUpdates))
-	for k := range h.draftUpdates {
-		keys = append(keys, k)
+	keys := make(programs.ContractUpdateKeyList, 0, len(updater.draftUpdates))
+	updates := make(programs.ContractUpdateList, 0, len(updater.draftUpdates))
+	for key, update := range updater.draftUpdates {
+		keys = append(keys, key)
+		updates = append(updates, update)
 	}
 
 	sort.Sort(keys)
-	return keys
+	sort.Sort(updates)
```

Comment:
You may consider using: [golang.org/x/exp/slices.SortFunc](https://pkg.go.dev/golang.org/x/exp/slices#SortFunc) since it has a couple of benefits:
* it only requires passing a sort function, which would allow us to dedup the `Less` method (and likely remove `Len` and `Swap`)
* it is generic

Reply:
that's a reasonable ask.  i was just being lazy.  will make the recommended change

Reply:
i've moved the sorter back into handler/contract.go since it's not safe to assume we can sort the key/update pairs together in general.

generic is overrated in this case.  your closure function will cost you whatever you've saved.